File: src/PythonClient/lib/quixstreaming/models/autooffsetreset.py
from enum import Enum
from .. import __importnet
import Quix.Sdk.Process


class AutoOffsetReset(Enum):
    Latest = 0,
    """
    Latest, starts from newest message if there is no stored offset
    """

    Earliest = 1,
    """
    Earliest, starts from the oldest message if there is no stored offset
    """

    Error = 2
    """
    Error, throws exception if there is no stored offset
    """

    @staticmethod
    def convert_from_net(net_enum: Quix.Sdk.Process.Kafka.AutoOffsetReset):
        if net_enum == Quix.Sdk.Process.Kafka.AutoOffsetReset.Latest:
            return AutoOffsetReset.Latest
        if net_enum == Quix.Sdk.Process.Kafka.AutoOffsetReset.Earliest:
            return AutoOffsetReset.Earliest
        if net_enum == Quix.Sdk.Process.Kafka.AutoOffsetReset.Error:
            return AutoOffsetReset.Error
        raise Exception('An unhandled AutoOffsetReset (' + str(net_enum) + ')')

    def convert_to_net(self):
        if self == AutoOffsetReset.Latest:
            return Quix.Sdk.Process.Kafka.AutoOffsetReset.Latest
        if self == AutoOffsetReset.Earliest:
            return Quix.Sdk.Process.Kafka.AutoOffsetReset.Earliest
        if self == AutoOffsetReset.Error:
            return Quix.Sdk.Process.Kafka.AutoOffsetReset.Error
        raise Exception('An unhandled AutoOffsetReset (' + str(self) + ')')

File: src/CsharpClient/QuixStreams.Telemetry/Kafka/AutoOffsetReset.cs
using System;

namespace QuixStreams.Telemetry.Kafka
{
    /// <summary>The offset to use when there is no saved offset for the consumer group.</summary>
    public enum AutoOffsetReset
    {
        /// <summary>Latest, starts from newest message if there is no stored offset</summary>
        Latest = 0,
        /// <summary>Earliest, starts from the oldest message if there is no stored offset</summary>
        Earliest = 1,
        /// <summary>Error, throws exception if there is no stored offset</summary>
        Error = 2
    }

    /// <summary>
    /// Confluent Kafka types extension methods
    /// </summary>
    public static partial class ConfluentTypeConverter
    {
        /// <summary>
        /// Convert Quix AutoOffsetReset type into Confluent Kafka AutoOffsetReset type
        /// </summary>
        /// <param name="offsetReset">Quix AutoOffsetReset type</param>
        /// <returns>Confluent Kafka AutoOffsetReset type</returns>
        public static Confluent.Kafka.AutoOffsetReset ConvertToKafka(this AutoOffsetReset offsetReset)
        {
            switch (offsetReset)
            {
                case AutoOffsetReset.Latest:
                    return Confluent.Kafka.AutoOffsetReset.Latest;
                case AutoOffsetReset.Earliest:
                    return Confluent.Kafka.AutoOffsetReset.Earliest;
                case AutoOffsetReset.Error:
                    return Confluent.Kafka.AutoOffsetReset.Error;
                default:
                    throw new ArgumentOutOfRangeException(nameof(offsetReset), offsetReset, null);
            }
        }
    }
}